pkg/gadgets/tcptop/tracer/tracer.go

```diff
@@ -0,0 +1,278 @@
+//go:build linux
+// +build linux
+
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this bio except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/tcptop/types"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+)
+
+// #include <linux/types.h>
+// #include "./bpf/tcptop.h"
+// #include <arpa/inet.h>
+// #include <stdlib.h>
+//
+//char *string_ip(const void *addr, int ip_type) {
+//	socklen_t size;
+//	char *ip;
+//
+//  // Should not occur because eBPF code already filter on this.
+//	if (ip_type != AF_INET && ip_type != AF_INET6)
+//		return NULL;
+//
+//	size = sizeof *ip * INET6_ADDRSTRLEN;
```

Comment:
sizeof *ip -> sizeof(*ip) ?

Reply:
Both are working, `sizeof` is a keyword, not a function.
And when I can use my C coding style, I prefer to not use the brackets.

Reply:
> I can use my C coding style

Well, it's not about your code style but about the project code style. Why to mess up it?

Reply:
Is there any agreement on coding style regarding C for `inspektor-gadget`?
If so, I will follow it, otherwise I thought I had the liberty to use mine.